libhoney.go

```diff
@@ -120,6 +120,30 @@ type Config struct {
 	Transport http.RoundTripper
 }
 
+// VerifyWriteKey calls out to the Honeycomb API to validate the write key, so
+// we can exit immediately if desired instead of happily sending events that
+// are all rejected.
+func VerifyWriteKey() error {
```

Comment:
This should either accept a `Config` argument, or be a method on the `Builder` type, instead of relying on global state.

Reply:
+1 passing in a `Config` argument, to allow you to use it independent of the rest of `libhoney`